src/backend/distributed/commands/utility_hook.c

```diff
@@ -176,6 +176,20 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 	}
 #endif
 
+	if (IsA(parsetree, VariableSetStmt))
+	{
+		VariableSetStmt *setStmt = (VariableSetStmt *) parsetree;
+
+		/* we don't process anything other than SET LOCAL */
+		/* need to be in a transaction to propagate SET LOCAL */
+		if (setStmt->is_local && IsMultiStatementTransaction())
+		{
+			ProcessVariableSetStmt(setStmt, queryString);
+		}
+
+		/* TODO: warn if not possible to propagate? */
```

Comment:
I guess we probably shouldn't emit anything by default, since this potentially happens very frequently.

Reply:
I'll remove the `TODO`, but I'm _mostly_ worried about the case where a user is modifying a GUC that's part of the whitelist (admittedly not yet part of the PR when you looked at it) but _isn't_ doing so in a transaction so we don't propagate its effects. That seems like it might be counterintuitive enough for us to warn them.